common/environment/environment.cpp

```diff
@@ -1157,11 +1302,66 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByComputer(const char * compu
     if (!computer)
         return NULL;
     buildDropZoneCache();
-    VStringBuffer xpath("Software/DropZone[@computer=\"%s\"]", computer);
+
+    StringBuffer x("@");
+    x.append(computer).append(NUMBER_OF_DROPZONES_SUFFIX);
+    unsigned numOfDropZoneByComputer = numOfDropzonesByComputer->getPropInt(x, -1);
+    if ( numOfDropZoneByComputer == -1 )
+        return NULL;
+
+    StringBuffer xpath("Software/DropZone[@computer=\"");
+    xpath.append(computer);
+    xpath.append(DROPZONE_BY_MACHINE_SUFFIX).append(DEFAULT_DROPZONE_INDEX);
+    xpath.append("\"]");
+
+    synchronized procedure(safeCache);
+    return (CConstDropZoneInfo *) getCache(xpath.str());
+}
+
+IConstDropZoneInfo * CLocalEnvironment::getDropZoneByComputer(const char * computer, const char * dzname) const
```

Comment:
am not sure when this would be used?
It would be an environment definition error if >1 DZ was defined with the same name.